### PR TITLE
Tweaks rockernaut spawn rate. Adds vault generation for asteroids

### PR DESCRIPTION
**Pull request by MadmanMartian**
Rockernauts are now much rarer. 

Adds asteroid vault generation, much like the vault generation (Very very much like the vault generation). PoC is a small arena where you can encounter the rockernaut boss, Angie.

### CODE REVIEW

**Comment by SonixApache** [comment]
a to mic

**Comment by DeityLink** [comment]
So can we replace the old boring asteroid secret rooms with carefully handcrafted and thematically polished vaults?

**Comment by MadmanMartian** [comment]
Yes

File: code/modules/randomMaps/vaults.dm [hunk 1]
```diff
@@ -84,7 +84,20 @@
...
+	var/surprise_number = rand(1, min(list_of_surprises.len, max_secret_rooms))
+
+	var/result = populate_area_with_vaults(/area/mine/unexplored, list_of_surprises, surprise_number, filter_function=/proc/asteroid_can_be_placed)
+
+	message_admins("<span class='info'>Loaded [result] out of [surprise_number] mining surprises.</span>")
+
+/proc/asteroid_can_be_placed(var/datum/map_element/E, var/turf/start_turf)
+	var/list/dimensions = E.get_dimensions()
+	var/result = check_complex_placement(start_turf,dimensions[1], dimensions[2])
+	to_chat(world, "placed at returns [result] at [start_turf] [start_turf.x] [start_turf.y] [start_turf.z]")
```
Comment (DamianX) [inline code/modules/randomMaps/vaults.dm:99]: nice

Comment (MadmanMartian) [reply code/modules/randomMaps/vaults.dm:99]: Nice

**Pull request merged**